.github/workflows/pr_test_helm_chart.yml

```
@@ -0,0 +1,107 @@
+name: Test Helm Charts on KOF
+
+on:
+  pull_request_target:
+    types:
+      - labeled
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - main
+      - release-*
+    paths:
+      - "promxy-operator/**"
+      - "charts/**"
+      - "!**.md"
+
+env:
+  GO_VERSION: "1.23"
+
+jobs:
+  build:
+    concurrency:
+      group: build-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    name: Build and Unit Test
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.vars.outputs.version }}
+      pr: ${{ steps.pr.outputs.result }}
+    permissions:
+      packages: write
+    steps:
+      - name: Get PR ref
+        uses: actions/github-script@v7
+        id: pr
+        with:
+          script: |
+            const { data: pullRequest } = await github.rest.pulls.get({
+              ...context.repo,
+              pull_number: context.payload.pull_request.number,
+            });
+            return pullRequest
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{fromJSON(steps.pr.outputs.result).merge_commit_sha}}
+          fetch-depth: 0
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: false
+
+  setup-kind-cluster:
+    name: Setup Kind Cluster
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Install Kind
+        run: |
+          curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.23.0/kind-linux-amd64
+          chmod +x ./kind
+          sudo mv ./kind /usr/local/bin/kind
+      - name: Create Kind Cluster
+        run: |
+          kind create cluster --name kof-cluster
+
+  setup-kof-and-kcm:
+    name: Install and Apply KCM, Then Deploy KOF
+    runs-on: ubuntu-latest
+    needs: setup-kind-cluster
+    steps:
+      - name: Checkout KCM repository
+        uses: actions/checkout@v4
+        with:
+          repository: k0rdent/kcm
+          ref: main
+          path: kcm-repo
+
+      - name: Install KCM CLI
+        run: |
+          make -C kcm-repo cli-install  # Runs `make cli-install` inside `kcm-repo`
+
+      - name: Apply KCM Dev Configuration
+        run: |
+          make -C kcm-repo dev-apply  # Runs `make dev-apply` inside `kcm-repo`
```

Comments are not needed here as `man make` shows what `-C` is, it is not some unique knowledge.
```suggestion
          make -C kcm-repo cli-install

      - name: Apply KCM Dev Configuration
        run: |
          make -C kcm-repo dev-apply
```